packages/phishing-controller/src/utils.ts

```diff
@@ -81,3 +88,136 @@ export const applyDiffs = (
     lastUpdated: latestDiffTimestamp,
   };
 };
+
+/**
+ * Validates the configuration object for the phishing detector.
+ *
+ * @param config - the configuration object to validate.
+ * @returns true if the configuration is valid.
+ * @throws an error if the configuration is invalid.
+ */
+export const validateConfig = (
+  config: unknown,
+): config is PhishingListState => {
+  if (config === null || typeof config !== 'object') {
+    throw new Error('Invalid config');
+  }
+
+  if ('tolerance' in config && !('fuzzylist' in config)) {
+    throw new Error('Fuzzylist tolerance provided without fuzzylist');
+  }
+
+  if (
+    'name' in config &&
+    (typeof config.name !== 'string' || config.name === '')
+  ) {
+    throw new Error("Invalid config parameter: 'name'");
+  }
+
+  if (
+    'version' in config &&
+    (!['number', 'string'].includes(typeof config.version) ||
+      config.version === '')
+  ) {
+    throw new Error("Invalid config parameter: 'version'");
+  }
+
+  return true;
+};
+
+/**
+ * Converts a domain string to a list of domain parts.
+ *
+ * @param domain - the domain string to convert.
+ * @returns the list of domain parts.
+ */
+export const domainToParts = (domain: string) => {
+  try {
+    return domain.split('.').reverse();
+  } catch (e) {
+    throw new Error(JSON.stringify(domain));
+  }
+};
+
+/**
+ * Converts a list of domain strings to a list of domain parts.
+ *
+ * @param list - the list of domain strings to convert.
+ * @returns the list of domain parts.
+ */
+export const processDomainList = (list: string[]) => {
+  return list.map(domainToParts);
+};
+
+/**
+ * Gets the default phishing detector configuration.
+ *
+ * @param override - the optional override for the configuration.
+ * @param override.allowlist - the optional allowlist to override.
+ * @param override.blocklist - the optional blocklist to override.
+ * @param override.fuzzylist - the optional fuzzylist to override.
+ * @param override.tolerance - the optional tolerance to override.
+ * @returns the default phishing detector configuration.
+ */
+export const getDefaultPhishingDetectorConfig = (override?: {
+  allowlist?: string[];
+  blocklist?: string[];
+  fuzzylist?: string[];
+  tolerance?: number;
+}): PhishingDetectorConfiguration => ({
+  allowlist: processDomainList(override?.allowlist || []),
+  blocklist: processDomainList(override?.blocklist || []),
+  fuzzylist: processDomainList(override?.fuzzylist || []),
+  tolerance: override?.tolerance || DEFAULT_TOLERANCE,
+});
```

Comment:
Yes, the idea was to give default values to each property, in case a consumer would override only one of them

Reply:
> Then the defaults for the rest of the fields will not be used with this approach, I think, as they only get filled in when the entire argument is missing?

No, if you pass `{ allowlist: [] }`, then it would fill in `[]` for `blocklist` and `fuzzylist` and `DEFAULT_TOLERANCE` for `tolerance`.